tests/py/test_billing_payday.py

```diff
@@ -568,6 +492,125 @@ def test_payin_dumps_transfers_for_debugging(self, cch, fch):
+        self.make_member('crusher', 150)
+        self.make_member('bruiser', 250)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_processes_takes_again(self):
+        # This catches a bug where payday_takes.amount was an int!
+        self.make_member('crusher', D('0.01'))
+        self.make_member('bruiser', D('5.37'))
+        self.run_through_takes()
+        assert P('crusher').balance == D('0.01')
+        assert P('bruiser').balance == D('5.37')
+        assert P('picard').balance  == D('0.00')
+
+    def test_pt_ignores_takes_set_after_the_start_of_payday(self):
+        self.make_member('crusher', 150)
+        self.start_payday()
+        self.make_member('bruiser', 250)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_ignores_takes_that_have_already_been_processed(self):
+        self.make_member('crusher', 150)
+        self.start_payday()
+        self.make_member('bruiser', 250)
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_clips_to_available(self):
+        self.make_member('alice', 350)
+        self.make_member('bruiser', 250)
+        self.make_member('crusher', 150)
+        self.make_member('zorro', 450)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('alice').balance   == D('100.00')
+        assert P('zorro').balance   == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_clips_to_balance_when_less_than_available(self):
+        self.set_available(1000)
+        self.make_member('alice', 350)
+        self.make_member('bruiser', 250)
+        self.make_member('crusher', 150)
+        self.make_member('zorro', 450)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('alice').balance   == D('137.00')
+        assert P('zorro').balance   == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_is_NOT_happy_to_deal_the_owner_in(self):
+        self.make_member('crusher', 150)
+        self.make_member('bruiser', 250)
+        self.enterprise.set_take_for(self.picard, D('0.01'), self.picard)
+        with pytest.raises(NotAllowed):
+            self.enterprise.set_take_for(self.picard, 50, self.picard)
+        return  # XXX allow owners to join teams!
```

Comment:
I know @kaguillera and I talked about this, and I thought I recorded somewhere that we thought it was okay to allow owners to join teams. Looking for that ...